### PR TITLE
chore(resources,common,developer,core): ldml update out of techpreview 🙀

### PR DESCRIPTION
**Pull request by srl295**
- includes patch for bn.xml
- pre-release ldml
- update to cldr 40fa7fd5b074a203581d04ca0c0b004ab9aa491b (main)

Fixes: #10900 
Fixes: #10901

_(Review note: split into 4 commits by functional area)_

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/types/build.sh [hunk 1]
```diff
@@ -29,8 +29,8 @@ builder_parse "$@"
 function compile_schemas() {
   # We need the schema files at runtime and bundled, so always copy it for all actions except `clean`
   local schemas=(
-    "$KEYMAN_ROOT/resources/standards-data/ldml-keyboards/techpreview/ldml-keyboard3.schema.json"
-    "$KEYMAN_ROOT/resources/standards-data/ldml-keyboards/techpreview/ldml-keyboardtest3.schema.json"
+    "$KEYMAN_ROOT/resources/standards-data/ldml-keyboards/45/ldml-keyboard3.schema.json"
+    "$KEYMAN_ROOT/resources/standards-data/ldml-keyboards/45/ldml-keyboardtest3.schema.json"
```
Comment (srl295) [inline common/web/types/build.sh:33]: 100% kicking the 'what about 46' can down the road here

**Comment by mcdurdin** [comment]
9 failing tests:

```
12:50:29     125 passing (950ms)
12:50:29     1 pending
12:50:29     9 failing
12:50:29   
12:50:29     1) disp
12:50:28          should fail to compile: sections/disp/invalid-var.xml:
12:50:28   
12:50:28         expected errors to be included
12:50:28         + expected - actual
12:50:28   
12:50:28          [
12:50:28            {
12:50:28         -    "code": 3149825
12:50:28         +    "code": 3145756
12:50:28              "detail": [undefined]
12:50:28         -    "message": "Error validating LDML XML file: /keyboard3: required: must have required property 'conformsTo' missingProperty=\"conformsTo\""
12:50:28         +    "message": "Reference to undefined string variable: ${missingdisplay}"
12:50:28            }
12:50:28         +  {
12:50:28         +    "code": 3145756
12:50:28         +    "detail": [undefined]
12:50:28         +    "message": "Reference to undefined string variable: ${missingoutput}"
12:50:28         +  }
12:50:28          ]
12:50:28   
12:50:28         at Context.<anonymous> (file:///C:/BuildAgent/work/7ac43416c45637e9/keyman/developer/src/kmc-ldml/test/helpers/index.ts:250:16)
12:50:28   
12:50:28     2) keys.kmap
12:50:28          should fail to compile: sections/keys/invalid-undefined-var-1.xml:
12:50:28   
12:50:28         expected errors to be included
12:50:28         + expected - actual
12:50:28   
12:50:28          [
12:50:28            {
12:50:28         -    "code": 3149825
12:50:28         +    "code": 3145756
12:50:28              "detail": [undefined]
12:50:28         -    "message": "Error validating LDML XML file: /keyboard3: required: must have required property 'conformsTo' missingProperty=\"conformsTo\""
12:50:28         +    "message": "Reference to undefined string variable: ${varsok}"
12:50:28            }
12:50:28          ]
12:50:28   
12:50:28         at Context.<anonymous> (file:///C:/BuildAgent/work/7ac43416c45637e9/keyman/developer/src/kmc-ldml/test/helpers/index.ts:250:16)
12:50:28   
12:50:28     3) keys.kmap
12:50:28          should fail to compile: sections/keys/invalid-undefined-var-1b.xml:
12:50:28   
12:50:28         expected errors to be included
12:50:28         + expected - actual
12:50:28   
12:50:28          [
12:50:28            {
12:50:28         -    "code": 3149825
12:50:28         +    "code": 3145756
12:50:28              "detail": [undefined]
12:50:28         -    "message": "Error validating LDML XML file: /keyboard3: required: must have required property 'conformsTo' missingProperty=\"conformsTo\""
12:50:28         +    "message": "Reference to undefined string variable: ${varsok}"
12:50:28            }
12:50:28          ]
12:50:28   
12:50:28         at Context.<anonymous> (file:///C:/BuildAgent/work/7ac43416c45637e9/keyman/developer/src/kmc-ldml/test/helpers/index.ts:250:16)
12:50:28   
12:50:28     4) tran
12:50:28          should fail to compile: sections/tran/fail-missing-var-1.xml:
12:50:28   
12:50:28         expected errors to be included
12:50:28         + expected - actual
12:50:28   
12:50:28          [
12:50:28            {
12:50:28         -    "code": 3149825
12:50:28         +    "code": 3145756
12:50:28              "detail": [undefined]
12:50:28         -    "message": "Error validating LDML XML file: /keyboard3: required: must have required property 'conformsTo' missingProperty=\"conformsTo\""
12:50:28         +    "message": "Reference to undefined string variable: ${missingfrom}"
12:50:28            }
12:50:28          ]
12:50:28   
12:50:28         at Context.<anonymous> (file:///C:/BuildAgent/work/7ac43416c45637e9/keyman/developer/src/kmc-ldml/test/helpers/index.ts:250:16)
12:50:28   
12:50:28     5) tran
12:50:28          should fail to compile: sections/tran/fail-missing-var-2.xml:
12:50:28   
12:50:28         expected errors to be included
12:50:28         + expected - actual
12:50:28   
12:50:28          [
12:50:28            {
12:50:28         -    "code": 3149825
12:50:28         +    "code": 3145756
12:50:28              "detail": [undefined]
12:50:28         -    "message": "Error validating LDML XML file: /keyboard3: required: must have required property 'conformsTo' missingProperty=\"conformsTo\""
12:50:28         +    "message": "Reference to undefined string variable: ${missingto}"
12:50:28            }
12:50:28          ]
12:50:28   
12:50:28         at Context.<anonymous> (file:///C:/BuildAgent/work/7ac43416c45637e9/keyman/developer/src/kmc-ldml/test/helpers/index.ts:250:16)
12:50:28   
12:50:28     6) tran
12:50:28          should fail to compile: sections/tran/fail-missing-var-3.xml:
12:50:28   
12:50:28         expected errors to be included
12:50:28         + expected - actual
12:50:28   
12:50:28          [
12:50:28            {
12:50:28         -    "code": 3149825
12:50:28         +    "code": 3145757
12:50:28              "detail": [undefined]
12:50:28         -    "message": "Error validating LDML XML file: /keyboard3: required: must have required property 'conformsTo' missingProperty=\"conformsTo\""
12:50:28         +    "message": "Reference to undefined set variable: $[missingset]"
12:50:28            }
12:50:28          ]
12:50:28   
12:50:28         at Context.<anonymous> (file:///C:/BuildAgent/work/7ac43416c45637e9/keyman/developer/src/kmc-ldml/test/helpers/index.ts:250:16)
12:50:28   
12:50:28     7) tran
12:50:28          should fail to compile: sections/tran/fail-missing-var-4.xml:
12:50:28   
12:50:28         expected errors to be included
12:50:28         + expected - actual
12:50:28   
12:50:28          [
12:50:28            {
12:50:28         -    "code": 3149825
12:50:28         +    "code": 3145757
12:50:28              "detail": [undefined]
12:50:28         -    "message": "Error validating LDML XML file: /keyboard3: required: must have required property 'conformsTo' missingProperty=\"conformsTo\""
12:50:28         +    "message": "Reference to undefined set variable: $[missingset]"
12:50:28            }
12:50:28          ]
12:50:28   
12:50:28         at Context.<anonymous> (file:///C:/BuildAgent/work/7ac43416c45637e9/keyman/developer/src/kmc-ldml/test/helpers/index.ts:250:16)
12:50:28   
12:50:28     8) tran
12:50:28          should fail to compile: sections/tran/fail-missing-var-5.xml:
12:50:28   
12:50:28         expected errors to be included
12:50:28         + expected - actual
12:50:28   
12:50:28          [
12:50:28            {
12:50:28         -    "code": 3149825
12:50:28         +    "code": 3145757
12:50:28              "detail": [undefined]
12:50:28         -    "message": "Error validating LDML XML file: /keyboard3: required: must have required property 'conformsTo' missingProperty=\"conformsTo\""
12:50:28         +    "message": "Reference to undefined set variable: $[missingset]"
12:50:28            }
12:50:28          ]
12:50:28   
12:50:28         at Context.<anonymous> (file:///C:/BuildAgent/work/7ac43416c45637e9/keyman/developer/src/kmc-ldml/test/helpers/index.ts:250:16)
12:50:28   
12:50:28     9) tran
12:50:28          should fail to compile: sections/tran/fail-missing-var-6.xml:
12:50:28   
12:50:28         expected errors to be included
12:50:28         + expected - actual
12:50:28   
12:50:28          [
12:50:28            {
12:50:28         -    "code": 3149825
12:50:28         +    "code": 3145756
12:50:28              "detail": [undefined]
12:50:28         -    "message": "Error validating LDML XML file: /keyboard3: required: must have required property 'conformsTo' missingProperty=\"conformsTo\""
12:50:28         +    "message": "Reference to undefined string variable: ${missingstr}"
12:50:28            }
12:50:28          ]
12:50:28   
12:50:28         at Context.<anonymous> (file:///C:/BuildAgent/work/7ac43416c45637e9/keyman/developer/src/kmc-ldml/test/helpers/index.ts:250:16)
12:50:28   
12:50:28   
```

File: developer/src/kmconvert/Keyman.Developer.System.LdmlKeyboardProjectTemplate.pas [hunk 2]
```diff
@@ -79,8 +79,8 @@ function TLDMLKeyboardProjectTemplate.GetKeyboardFilename: string;
     //  '<!DOCTYPE keyboard3 SYSTEM "ldmlKeyboard3.dtd">'#13#10+
     '<keyboard3>'#13#10+
```
Comment (mcdurdin) [inline developer/src/kmconvert/Keyman.Developer.System.LdmlKeyboardProjectTemplate.pas:80]:
```suggestion
    '<keyboard3 xmlns="https://schemas.unicode.org/cldr/45/keyboard3">'#13#10+
```

Need to also fix line 133, still has 'techpreview':

```pas
  root.Attributes['conformsTo'] := 'techpreview';
```

**Review by mcdurdin — CHANGES_REQUESTED** [review]
Apart from the failing tests, a small update to kmconvert should also address #10901.

**Comment by srl295** [comment]
hrm, i think on the third try of generating this i ran kmc/build.sh and missed kmc-ldml/build.sh..

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.283-beta](https://keyman.com/downloads/releases/beta/17.0.283)

**Pull request merged**